feature-charity/src/main/java/com/hedvig/android/feature/charity/CharityActivity.kt

```diff
@@ -0,0 +1,38 @@
+package com.hedvig.android.feature.charity
+
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.material3.windowsizeclass.ExperimentalMaterial3WindowSizeClassApi
+import androidx.compose.material3.windowsizeclass.calculateWindowSizeClass
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.core.view.WindowCompat
+import coil.ImageLoader
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.feature.charity.ui.CharityScreen
+import org.koin.android.ext.android.get
+import org.koin.androidx.viewmodel.ext.android.getViewModel
+
+@OptIn(ExperimentalMaterial3WindowSizeClassApi::class)
+class CharityActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    WindowCompat.setDecorFitsSystemWindows(window, false)
+    val imageLoader: ImageLoader = get()
+    val viewModel = getViewModel<CharityViewModel>()
```

Comment:
What's the reason for getting these dependencies with koin in the `onCreate`-body over the more common `inject`-variant?

Reply:
Reducing the scope in which it's available mostly. If it doesn't need to be accessed globally in this class and can stay contained inside the function I figured it'd better stay that way.
The lazy impl in koin is just doing a `lazy` wrapper around `get()` and the `lazy` comes with an overhead of doing all this fetching by aquring a lock to make sure it's thread safe and that kind of stuff https://github.com/InsertKoinIO/koin/blob/c954e53959cf982725ca33c9680acb650c63da52/android/koin-android/src/main/java/org/koin/android/ext/android/ComponentCallbackExt.kt#L40-L58 
So figured why not get the benefit of removing this locking and the reduced scope, seemed like a win-win to me.